easybuild/main.py

```diff
@@ -731,10 +736,29 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
         cleanup(logfile, eb_tmpdir, testing, silent=False)
 
 
+def prepare_main(args=None, logfile=None, testing=None):
+    """
+    Prepare Main: Set up eb_go and consequently, cfg_settings in order to obtain the hooks,
+    and prevent the configuration being set up twice.
```

Comment:
This doesn't prevent anything, and it's not specific to hooks in any way.

I would just use: `Prepare for calling main function by setting up the EasyBuild configuration.`